x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/Attribute.java

```diff
@@ -25,7 +25,7 @@
  * is a named expression (an {@code Alias} will be created automatically for it).
  * The rest are not as they are not part of the projection and thus are not part of the derived table.
  */
-public abstract class Attribute extends NamedExpression {
+public abstract class Attribute extends NamedExpression implements Comparable<Attribute>{
```

Comment:
Making generic class Attribute to implement Comparable just for the purposes of constructing a nice error message is a bit too much. Please remove the comparison implementation from here and use a custom comparator to sort the ambiguous attributes.